modules/frontdoor_classic/variables.tf

```diff
@@ -0,0 +1,112 @@
+variable "location" {
+  type        = string
+  description = "A datacenter location in Azure."
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Name of the resource group."
+}
+
+variable "name" {
+  type        = string
+  description = "Specifies the name of the Front Door service."
+}
+
+variable "endpoints" {
+  type = list(object({
+    name                                    = string
+    host_name                               = string
+    session_affinity_enabled                = optional(bool)
+    session_affinity_ttl_seconds            = optional(number)
+    web_application_firewall_policy_link_id = optional(string)
+    custom_https                            = optional(bool)
+  }))
+  description = "List of frontend_endpoint objects for frontdoor. At least one object is required."
+}
+
+variable "routes" {
+  type = list(object({
+    name               = string
+    frontend_endpoints = list(string)
+    enabled            = optional(bool)
+    accepted_protocols = optional(list(string))
+    patterns_to_match  = optional(list(string))
+    redirect_configuration = optional(object({
+      redirect_type       = string
+      redirect_protocol   = optional(string)
+      custom_host         = optional(string)
+      custom_fragment     = optional(string)
+      custom_path         = optional(string)
+      custom_query_string = optional(string)
+    }))
+    forwarding_configuration = optional(object({
+      backend_pool_name                     = string
+      cache_enabled                         = optional(bool)
+      cache_use_dynamic_compression         = optional(bool)
+      cache_query_parameter_strip_directive = optional(string)
+      cache_query_parameters                = optional(list(string))
+      cache_duration                        = optional(string)
+      custom_forwarding_path                = optional(string)
+      forwarding_protocol                   = optional(string)
+    }))
+  }))
+  description = "A list of routing_rule objects for frontdoor. At least one object is required."
```

Comment:
If atleast one is required, could this be validated by making sure the length of this list is > 0 (validation block)? Just to make sure a [] isn't used as input.

The same goes for the other ones where one is required

Reply:
Smart! It will give an error, but validation errors can be way more specific.